lib/ansible/galaxy/role.py

```diff
@@ -303,54 +362,16 @@ def install(self):
                     except Exception:
                         raise AnsibleError("this role does not appear to have a valid meta/main.yml file.")
 
-                # we strip off any higher-level directories for all of the files contained within
-                # the tar file here. The default is 'github_repo-target'. Gerrit instances, on the other
-                # hand, does not have a parent directory at all.
-                installed = False
-                while not installed:
-                    display.display("- extracting %s to %s" % (self.name, self.path))
+                for path in self.paths:
+                    self.path = path
```

Comment:
Instead of assigning an instance attribute, just pass it to the `self._extract()` call. This way, that function wouldn't rely on some magical state that must always be preset correctly outside it. The better API is to let it work with `path` as a locally-available value.